packages/common/src/services/audius-backend/solana.ts

```diff
@@ -757,3 +713,210 @@ export const createVersionedTransaction = async (
+  )
+  transaction.sign([sender])
+  const signature = await sdk.services.paymentRouterClient.sendTransaction(
+    transaction,
+    { skipPreflight: true }
+  )
+  return signature
+}
+
+/**
+ * Transfers tokens out of a user bank.
+ * Notes:
+ * - Including a signer will mark this transfer as a "withdrawal preparation"
+ *   by signing a memo indicating such. This prevents the transfer from showing
+ *   as a withdrawal on the withdrawal history page.
+ * - Users have restrictions on creating token accounts via relay, so if the
+ *   destination token account doesn't exist this might fail.
+ */
+export const transferFromUserBank = async ({
+  sdk,
+  mint,
+  connection,
+  amount,
+  ethWallet,
+  destinationWallet,
+  track,
+  make,
+  analyticsFields,
+  signer
+}: {
+  sdk: AudiusSdk
+  mint: PublicKey
+  connection: Connection
+  amount: number
+  ethWallet: string
+  destinationWallet: PublicKey
+  track: CommonStoreContext['analytics']['track']
+  make: CommonStoreContext['analytics']['make']
+  analyticsFields: any
+  signer?: Keypair
+}) => {
+  let isCreatingTokenAccount = false
+  try {
+    const instructions: TransactionInstruction[] = []
+
+    const destination = getAssociatedTokenAddressSync(mint, destinationWallet)
+
+    try {
+      await getAccount(connection, destination)
+    } catch (e) {
+      // Throws if token account doesn't exist or account isn't a token account
+      isCreatingTokenAccount = true
+      console.debug(e)
+      console.debug(
+        'Associated token account',
+        destination.toBase58(),
+        'does not exist. Creating w/ transfer...'
+      )
+
+      // Historically, the token account was created in a separate transaction
+      // after swapping USDC to SOL via Jupiter and funded via the root wallet.
+      // This is no longer the case. Reusing the same Amplitude events anyway.
+      await track(
+        make({
+          eventName: Name.WITHDRAW_USDC_CREATE_DEST_TOKEN_ACCOUNT_START,
+          ...analyticsFields
+        })
+      )
+      const payerKey = await sdk.services.solanaRelay.getFeePayer()
+      const createAtaInstruction =
+        createAssociatedTokenAccountIdempotentInstruction(
+          payerKey,
+          destination,
+          destinationWallet,
+          mint
+        )
+      instructions.push(createAtaInstruction)
+    }
+
+    const secpTransferInstruction =
+      await sdk.services.claimableTokensClient.createTransferSecpInstruction({
+        auth: sdk.services.auth,
+        amount,
+        ethWallet,
+        mint,
+        destination,
+        instructionIndex: instructions.length
+      })
+    instructions.push(secpTransferInstruction)
+
+    const transferInstruction =
+      await sdk.services.claimableTokensClient.createTransferInstruction({
+        ethWallet,
+        mint,
+        destination
+      })
+    instructions.push(transferInstruction)
+
+    if (signer) {
```

Comment:
nit: feels a little odd that the way you specify if there should be a prepare memo is by passing a signer? It's kind of a hidden side-effect.

Reply:
agree... potentially could add a separate "memo" param alongside the optional signer. wdyt?

Reply:
I think that makes a little more sense. I'm struggling to remember if we actually need the prepare memo to be signed? Signing memos is optional and I'm not sure if we really need to verify this particular piece of information since it's just used as metadata when we index.